src/xdg.rs

```diff
@@ -74,25 +78,42 @@ impl DesktopEntries {
                     Err(_) => continue,
                 };
 
+                // Ignore all groups other than the `Desktop Entry` one.
+                //
+                // Since `Desktop Entry` must be the first group, we just stop at the next group
+                // header.
+                let lines = desktop_file.lines().take_while(|line| {
+                    line.trim_end() == "[Desktop Entry]" || !line.starts_with('[')
+                });
+
                 let mut icon = None;
                 let mut exec = None;
                 let mut name = None;
 
-                for line in desktop_file.lines() {
-                    if let Some(value) = line.strip_prefix("Name=") {
-                        name = Some(value.to_owned());
-                    } else if let Some(value) = line.strip_prefix("Icon=") {
-                        icon = desktop_entries.loader.load(value, icon_size).ok();
-                    } else if let Some(value) = line.strip_prefix("Exec=") {
-                        // Remove %f/%F/%u/%U/%k variables.
-                        let filtered = value
-                            .split(' ')
-                            .filter(|arg| !matches!(*arg, "%f" | "%F" | "%u" | "%U" | "%k"));
-                        exec = Some(filtered.collect::<Vec<_>>().join(" "));
-                    }
+                // Find name, icon, and executable for the desktop entry.
+                for line in lines {
+                    // Get K/V pairs, allowing for whitespace around the assignment operator.
+                    let (key, value) = match line.split_once('=') {
+                        Some((key, value)) => (key.trim_end(), value.trim_start()),
+                        None => continue,
+                    };
```

Comment:
Splitting here should ensure things like `Key = Value` works properly, since that's allowed according to the spec.